app/src/main/java/org/mifos/mobile/viewModels/RegistrationViewModel.kt

```diff
@@ -67,8 +69,13 @@ class RegistrationViewModel @Inject constructor(private val userAuthRepositoryIm
             if (response?.isSuccessful == true) {
                 _registrationUiState.value = RegistrationUiState.Success
             } else {
-                _registrationUiState.value =
-                    response?.errorBody()?.string()?.let { RegistrationUiState.Error(it) }
+                response?.code()?.compareTo(400)?.let {
+                    if (it >= 0) {
+                        _registrationUiState.value = RegistrationUiState.Error(R.string.could_not_register_user_error)
+                    }
+                    return@launch
```

Comment:
why are we using launch here ?

Reply:
To handle the edge case when no response is coming but code is 404, so that 2 popups are not shown

Reply:
If we remove the comparisons with 400 then we can skip it entirely

Reply:
changes done